src/osd/osd_types.h

```diff
@@ -4604,14 +4608,20 @@ struct object_manifest_t {
     TYPE_NONE = 0,
     TYPE_REDIRECT = 1, 
     TYPE_CHUNKED = 2, 
+    TYPE_DEDUPED = 3, 
+  };
+  enum {
+    TYPE_SHA1_FINGERPRINT = 1,
```

Comment:
This seems like the important addition.  It feels like a "policy" or preference, though: it does't tell us anything about the existing chunks (which may or may not use this hash, or any hash), only what we ought to do with any new chunks.

In fact, do we need to persist this at all?

Reply:
@liewegas This is post-dedup, which means the fingerprint is not generated when client's data is come. The fingerprint will be generated at flushing step. So, we need to keep fingerprint info such as sha1, sha256. 
Without fingerprint info, how do we know which fingerprint method is used?